### PR TITLE
Created Borrower Class, its fields' classes and BorrowerRecords Class

### PR DESCRIPTION
**Pull request by linyutinglyt**
Created class under borrower package and some tests for the fields.

### CODE REVIEW

**Comment by seeziyang** [comment]
Great job! Looks good!

File: src/main/java/seedu/address/model/borrower/Email.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's email in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidEmail(String)}
+ */
+public class Email {
+
+    private static final String SPECIAL_CHARACTERS = "!#$%&'*+/=?`{|}~^.-";
+    public static final String MESSAGE_CONSTRAINTS = "Emails should be of the format local-part@domain "
```
Comment (seeziyang) [inline src/main/java/seedu/address/model/borrower/Email.java:13]: 👍

**Comment by shiyao821** [comment]
Seems great! Good to Merge

File: src/main/java/seedu/address/model/BorrowerIdGenerator.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * Generates a new borrower ID based on the current borrower ID index.
+     */
+    public static BorrowerId generateBorrowerId() {
+        currentBorrowerIdIndex++;
+        int paddingLength = getPaddingLength();
+        String padding = getPadding(paddingLength);
+        BorrowerId id = new BorrowerId(PREFIX + padding + currentBorrowerIdIndex);
+        while (borrowers.checkIfBorrowerIdExists(id)) {
+            currentBorrowerIdIndex++;
```
Comment (hoholyin) [inline src/main/java/seedu/address/model/BorrowerIdGenerator.java:31]: You should update id here to ensure that the while loop does not do on infinitely.

File: src/main/java/seedu/address/model/borrower/Email.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.borrower;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's email in the address book.
```
Comment (hoholyin) [inline src/main/java/seedu/address/model/borrower/Email.java]: might want to update the comments to make it relevant to our project

File: src/main/java/seedu/address/model/borrower/Name.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.borrower;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's name in the address book.
```
Comment (hoholyin) [inline src/main/java/seedu/address/model/borrower/Name.java]: Update comments here as well

**Review by hoholyin — COMMENTED** [review]
Need to update the logic for BorrowerIdGenerator. Other than that, all looks good

**Pull request merged**